src/engine/threaded_engine.h

```diff
@@ -283,8 +308,20 @@ class ThreadedEngine : public Engine {
    */
   void ExecuteOprBlock(RunContext run_ctx, OprBlock *opr_block) {
     ThreadedOpr* threaded_opr = opr_block->opr;
+#if MXNET_USE_PROFILER
+    if (opr_block->profiling) {
+      const Context& ctx = opr_block->ctx;
+      opr_block->opr_stat = Profiler::Get()->AddOprStat(ctx.dev_type, ctx.dev_id);
+      uint64_t id = std::hash<std::thread::id>()(std::this_thread::get_id());
+      opr_block->opr_stat->thread_id = id;
+      opr_block->opr_stat->opr_name  =
+          (threaded_opr->opr_name ? std::string(threaded_opr->opr_name) : "unknown");
+      // record operator start timestamp
+      SetOprStart(opr_block->opr_stat);
+    }
+#endif
     CallbackOnComplete callback = this->CreateCallback(
-        ThreadedEngine::OnCompleteStatic, threaded_opr);
+        ThreadedEngine::OnCompleteStatic, opr_block);
```

Comment:
yeah, it's a right change, because I have to get opr_block->opr_stat, which have to put in opr_block instead of threaded_opr, when executing callback (https://github.com/dmlc/mxnet/pull/3163/files#diff-168fac476949a07835e3e19e64f72f2bR411)